python/tank/bootstrap/manager.py

```diff
@@ -57,25 +60,69 @@ def __init__(self, sg_user=None):
         # defaults
         self._bundle_cache_fallback_paths = []
         self._caching_policy = self.CACHE_SPARSE
-        self._pipeline_configuration_name = None
+        self._pipeline_configuration_identifier = None # name or id
         self._base_config_descriptor = None
         self._progress_cb = None
         self._do_shotgun_config_lookup = True
         self._plugin_id = None
+        self._pre_engine_start_callback = None
 
         log.debug("%s instantiated" % self)
 
-
     def __repr__(self):
+        if self._pipeline_configuration_identifier is None:
+            identifier_type = "is"
+        elif isinstance(self._pipeline_configuration_identifier, int):
+            identifier_type = "id"
+        else:
+            identifier_type = "name"
+
         repr  = "<TkManager "
         repr += " User %s\n" % self._sg_user
         repr += " Cache fallback path %s\n" % self._bundle_cache_fallback_paths
         repr += " Caching policy %s\n" % self._caching_policy
         repr += " Plugin id %s\n" % self._plugin_id
-        repr += " Config %s\n" % self._pipeline_configuration_name
+        repr += " Config %s %s\n" % (identifier_type, self._pipeline_configuration_identifier)
         repr += " Base %s >" % self._base_config_descriptor
         return repr
 
+    def get_pipeline_configurations(self, project):
+        """
+        Retrieves the pipeline configurations available for a given project.
+
+        It also takes into account the current user and optional pipeline_configuration name or id. If the
+        :method:``ToolkitManager.pipeline_configuration`` attribute has been set to a string, it will look
+        for pipeline configurations with that specific name. If it has been set to ``None``, any pipeline
+        that can be applied for the current user and project will be retrieved. Note that this method does
+        not support ``ToolkitManager.pipeline_configuration`` being an integer.
+
+        :param project: Project entity link to enumerate pipeline configurations for. If ``None``, this will enumerate
+            the pipeline configurations for the site configuration.
+        :type project: Dictionary with keys ``type`` and ``id``, or ``None`` for the site
+
+        :returns: List of pipeline configurations.
+        :rtype: List of dictionaries with keys ``type``, ``id`` and ``name``.
+        """
+
+        if isinstance(self.pipeline_configuration, int):
+            raise TankBootstrapError("Can't enumerate pipeline configurations matching a specific id.")
```

Comment:
I remember having a very good reason for not wanting to support it, but can't remember. I should have documented it. Could we not support it for now and add it later if required?